### ISSUE TITLE
large_file_skip_char_limit not working

### ISSUE THREAD

**Issue by kieronellis** [sqlfluff/sqlfluff#3820]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Setting any value for `large_file_skip_char_limit` seems to be ignored when linting.

### Expected Behaviour

If it is set up like `large_file_skip_char_limit = 100` then all files with more than 100 characters should be skipped.

Or in our case, if a file is >20,000 characters, then setting the limit to `large_file_skip_char_limit = 0` should mean that this rule is ignored.

### Observed Behaviour

Regardless of what we change `large_file_skip_char_limit` to, it is ignored.

### How to reproduce

Set `large_file_skip_char_limit = 100` and run sqlfluff lint on a file with linting violations and more than 100 characters, the file should be skipped.

### Dialect

bigquery

### Version

1.2.1

### Configuration

.sqlfluff
```
# https://docs.sqlfluff.com/
[sqlfluff]
templater = dbt
dialect = bigquery

# Exclude:
# L008 Space after comma.
# L029 Keywords should not be used as identifiers.
# L032 Prefer specifying join keys instead of using USING.
# L034 Select wildcards then simple targets before calculations and aggregates.
# L048 Quoted literals should be surrounded by a single whitespace.
# L054 Inconsistent column references in GROUP BY/ORDER BY clauses.
exclude_rules = L008, L029, L032, L034, L048, L054
# Number of passes to run before admitting defeat, default = 10
runaway_limit = 100
large_file_skip_char_limit = 100

[sqlfluff:rules:L003]
# Indentation not consistent with previous lines.
tab_space_size = 2

[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = lower

[sqlfluff:rules:L014]
# Unquoted identifiers
extended_capitalisation_policy = lower

[sqlfluff:rules:L016]
max_line_length = 120
ignore_comment_clauses = True

[sqlfluff:rules:L019]
comma_style = trailing

[sqlfluff:rules:L030]
# Function names
extended_capitalisation_policy = lower

[sqlfluff:rules:L040]
# Null & Boolean Literals
capitalisation_policy = lower

[sqlfluff:rules:L063]
# Data Types
extended_capitalisation_policy = lower

[sqlfluff:rules:L064]
# Consistent usage of preferred quotes for quoted literals
preferred_quoted_literal_style = single_quotes
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
I've confirmed this issue 👍 

The configuration is being applied from the default config - and that is working - but not the configuration applied by the user. This is why the config from your `.sqlfluff` file isn't being applied. This applies both to the old `large_file_skip_char_limit` and to the newer `large_file_skip_byte_limit`.

**Comment by alanmcruickshank** [comment]
Running on the current master branch I can't recreate this issue.

With the following file `test_big.sql`:
```sql
SELECT some_long_filename
```

And the following `.sqlfluff` file:

```ini
[sqlfluff]
large_file_skip_char_limit = 10
```

The cli output is:

```sh
> sqlfluff lint -d ansi test_big.sql
WARNING    The config value large_file_skip_char_limit was found set. This feature will be removed in a future release, please use the more efficient 'large_file_skip_byte_limit' instead.
WARNING    Length of file 'test_big.sql' is over 10 characters. Skipping to avoid parser lock. Users can increase this limit in their config by setting the 'large_file_skip_char_limit' value, or disable by setting it to zero.
All Finished � �!
```

Likewise setting `large_file_skip_byte_limit` in the same way I get:
```sh
> sqlfluff lint -d ansi test_big.sql
WARNING    Length of file 'test_big.sql' is 25 bytes which is over the limit of 10 bytes. Skipping to avoid parser lock. Users can increase this limit in their config by setting the 'large_file_skip_byte_limit' value, or disable by setting it to zero.
All Finished � �!
```

Can you try upgrading to `1.3.0` and confirming whether you still have the same issue?

**Comment by fmms** [comment]
Might be related to #3693

**Comment by alanmcruickshank** [comment]
Very likely @fmms . @kieronellis - I would suggest using `large_file_skip_byte_limit` rather than `large_file_skip_char_limit` and upgrading to `1.3.1` which has much more robust support for both.

Can you confirm whether the issue is still occurring post upgrade?

**Comment by alanmcruickshank** [comment]
I'm going to close this issue as stale. I think it's been addressed.
